lib/delegate.rb

```diff
@@ -410,16 +410,16 @@ def __setobj__(obj)  # :nodoc:
       __raise__ ::ArgumentError, "cannot delegate to self" if self.equal?(obj)
       @delegate_dc_obj = obj
     end
+  end
+  klass.include(Module.new do
```

Comment:
I'm always a bit annoyed by anonymous modules. I wonder if it would be worth naming this module.

e.g. `klass.const_set("DelegationMethods", Module.new)`.

The benefit is that it makes it much clearer what is going on when introspecting `ancestors` or `method(:foo).owner`.

Of course there is the obvious risk of name clash etc, so maybe not worth it?

Reply:
> I'm always a bit annoyed by anonymous modules.

That's a good point.

> Of course there is the obvious risk of name clash etc, so maybe not worth it?

Perhaps a name that deviates from convention would be safe?  I've pushed a commit that uses  `DelegateClass_Methods`.